backend/matchmaking/models.py

```diff
@@ -1,36 +1,29 @@
 
-
-# Create your models here.
-# Do I need a to dict method even though there is a serializer?
 class FilmResults(models.Model):
     user = models.OneToOneField(CustomUser, on_delete=models.CASCADE, primary_key=True)
-    a_new_hope = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    empire_strikes_back = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    return_of_the_jedi = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    phantom_menace = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    attack_of_the_clones = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    revenge_of_the_sith = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    force_awakens = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    last_jedi = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    rise_of_skywalker = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    rogue_one = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    solo = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    mandalorian = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    book_of_boba_fett = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    obi_wan_kenobi = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    andor = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    ahsoka = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    clone_wars = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    rebels = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    resistance = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    bad_batch = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    visions = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    tales_of_the_empire = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-    tales_of_the_jedi = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)],blank=True)
-
-
-#class QuestionResults(models.Model):
-#    question1 = models.TextChoices()
\ No newline at end of file
+    a_new_hope = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(10)], default=1, blank=True)
```

Comment:
We decided that if a person had not seen the film then that film would not be used in the computation. Setting a default value of 1 means not seeing the film is the same as saying they hate it.

I don't think this is what we talked about or what the desired functionality is. But maybe you discussed this with others.

Reply:
I dont know how to do it. Maybe you could show us?

Reply:
This was done by the overlap dictionary. It used the truthiness of the values. the plan was to have the integer fields be nullable. Setting a default value messes with this. I think there was an issue with passing information appropriately from the front end to the back end.

The solution we have now sets a default value of -1. Then the function on both exist in match_users.py  is used to construct the dictionary overlap. In the for loop, we check against this dictionary to see if anything will be contributed.

It is a bit embarassing now to look at this. both_exist would be much cleaner if it just returned a>0 and b>0. Alas. Also, we could have just set the default value to None or Null on the back end. Or even just used a CharField and then converted to ints when doing the combination. Makes me feel stupid. ugh.

Reply:
👍